### PR TITLE
refactor: update collaboration copy to connection

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This PR updates all `collaboration` copy to `connection` related to https://github.com/open-sauced/app/pull/1831

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: migrations/20230517151100_add_user_collaborations.sql [hunk 1]
```diff
@@ -1,4 +1,4 @@
-create table if not exists public.user_collaborations
+create table if not exists public.user_connections
```
Comment (jpmcb) [inline migrations/20230517151100_add_user_collaborations.sql:1]: This needs it's own migration to transform the current `user_collaborations` table to a new `user_connections` table. We shouldn't modify existing migrations in place.

File: public/diagrams/README.md [hunk 2]
```diff
@@ -29,7 +29,7 @@
 | [public.user_top_repos](public.user_top_repos.md)                         | 6       |         | BASE TABLE |
 | [public.pg_stat_statements_info](public.pg_stat_statements_info.md)       | 2       |         | VIEW       |
 | [public.pg_stat_statements](public.pg_stat_statements.md)                 | 43      |         | VIEW       |
-| [public.user_collaborations](public.user_collaborations.md)               | 10      |         | BASE TABLE |
+| [public.user_connections](public.user_connections.md)                     | 10      |         | BASE TABLE |
```
Comment (jpmcb) [inline public/diagrams/README.md:32]: isn't everything in `public/diagrams` auto generated by the release action?

File: src/user-lists/entities/user-list-contributor-stats.entity.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { ApiModelProperty } from "@nestjs/swagger/dist/decorators/api-model-prop
 @Entity({ name: "user_list_contributors" })
 export class DbUserListContributorStat {
   @ApiModelProperty({
-    description: "User list collaborator's login",
+    description: "User list connection's login",
```
Comment (jpmcb) [inline src/user-lists/entities/user-list-contributor-stats.entity.ts:7]: This is fine but this description doesn't really (and probably didn't to begin with 🙈) make sense.

```suggestion
    description: "User list contributor's login",
```
might make more sense

File: src/user-lists/entities/user-list-contributor.entity.ts [hunk 4]
```diff
@@ -74,7 +74,7 @@ export class DbUserListContributor {
   public user_list_contributor!: DbUser;
 
   @ApiModelProperty({
-    description: "User list collaborator's login",
+    description: "User list connection's login",
```
Comment (jpmcb) [inline src/user-lists/entities/user-list-contributor.entity.ts:77]: Same here:

```suggestion
    description: "User list contributor's login",
```

**Review by jpmcb — CHANGES_REQUESTED** [review]

File: migrations/20230517151100_add_user_collaborations.sql (hunk 1 above)
Comment (OgDev-01) [reply migrations/20230517151100_add_user_collaborations.sql:1]: What are the steps to run the migrations?

**Comment by OgDev-01** [comment]
Closing this from a conclusion on slack chat https://opensaucedworkspace.slack.com/archives/C052YRD69JN/p1696864714090009

**Pull request closed without merging**